### PR TITLE
Allow deletion of placeholder users when permission of :manage_placeholder_user give

### PR DESCRIPTION
**Pull request by wielinde**
to :manage_user and :manage_placeholder_user

Todo:

- [x] Rename permission :add_user to :manage_user
- [x] Rename permission :add_placeholder_user to :manage_placeholder_user
- [x] Allow deletion of placeholder users when permission :manage_placeholder_user is present.
- [x] Add deletion info for placeholders
- [x] Extend specs 
- [x] Lock placeholder users when scheduling deleting and filter them in the list

### CODE REVIEW

File: app/cells/placeholder_users/row_cell.rb [hunk 1]
```diff
@@ -46,13 +46,11 @@ def button_links
     end
 
     def delete_link
-      return nil unless User.current.admin?
+      return nil unless User.current.admin? || PlaceholderUsers::DeleteContract.deletion_allowed?(User.current)
```
Comment (ulferts) [inline app/cells/placeholder_users/row_cell.rb]: I believe the `User.current.admin?` check to be superfluous

File: app/contracts/placeholder_users/delete_contract.rb [hunk 2]
```diff
@@ -39,19 +39,11 @@ class DeleteContract < ::DeleteContract
     #
     # @param actor [User] User who wants to delete the given placeholder user.
     def self.deletion_allowed?(actor)
-      actor.admin? && actor.active?
+      actor.admin? && actor.active? || actor.allowed_to_globally?(:manage_placeholder_user)
```
Comment (ulferts) [inline app/contracts/placeholder_users/delete_contract.rb]: This permission check can be shortened.

File: app/contracts/users/create_contract.rb [hunk 3]
```diff
@@ -56,7 +56,7 @@ def no_auth?
     ##
     # Users can only be created by Admins
     def user_allowed_to_add
-      unless user.admin? || user.allowed_to_globally?(:add_user)
+      unless user.admin? || user.allowed_to_globally?(:manage_user)
```
Comment (ulferts) [inline app/contracts/users/create_contract.rb]: See above

File: app/contracts/users/update_contract.rb [hunk 4]
```diff
@@ -37,10 +37,10 @@ class UpdateContract < BaseContract
     ##
     # Users can only be updated when
     # - the user is editing herself
-    # - the user has the global add_user CRU permission
+    # - the user has the global manage_user CRU permission
     # - the user is an admin
     def user_allowed_to_update
-      unless user == model || user.admin? || user.allowed_to_globally?(:add_user)
+      unless user == model || user.admin? || user.allowed_to_globally?(:manage_user)
```
Comment (ulferts) [inline app/contracts/users/update_contract.rb]: And again

File: app/services/placeholder_users/delete_service.rb [hunk 5]
```diff
@@ -30,6 +30,7 @@
 
 class PlaceholderUsers::DeleteService < ::BaseServices::Delete
   def destroy(placeholder)
+    placeholder.update_column(:status, Principal.statuses[:locked])
```
Comment (ulferts) [inline app/services/placeholder_users/delete_service.rb]: Using `placeholder.locked!` would be easier to understand.

**Review by ulferts — CHANGES_REQUESTED** [review]
Only very tiny issues. Should be good to merge soon 🎉 

Only one front end issue found.

In the deletion text the escaping is displayed:

image

File: app/services/placeholder_users/delete_service.rb (hunk 5 above)
Comment (oliverguenther) [reply app/services/placeholder_users/delete_service.rb]: Ah, these are autogenerated from the enum then? I only saw the ones in `user.rb`.

Comment (ulferts) [reply app/services/placeholder_users/delete_service.rb]: Yes, they are: https://api.rubyonrails.org/v6.1.2.1/classes/ActiveRecord/Enum.html

**Review by ulferts — APPROVED** [review]

**Pull request merged**